metaflow/runtime.py

```diff
@@ -798,56 +836,110 @@ def __init__(
 
                 # If _get_task_id returns True it means the task already existed, so
                 # we wait for it.
-                self._wait_for_clone = self._get_task_id(clone_task_id)
+                task_id_exists_already = self._get_task_id(clone_task_id)
+
+                # We may not have access to task datastore on first resume attempt, but
+                # on later resume attempt, we should check if the resume task is complete
+                # or not. This is to fix the issue where the resume leader was killed
+                # unexpectedly during cloning and never mark task complete.
+                try:
+                    task_completed = self.results["_task_ok"]
+                except DataException as e:
+                    pass
             else:
-                self._wait_for_clone = False
                 self._get_task_id(task_id)
 
             # Store the mapping from current_pathspec -> origin_pathspec which
             # will be useful for looking up origin_ds_set in find_origin_task.
             self.clone_pathspec_mapping[self._path] = origin.pathspec
             if self.step == "_parameters":
-                # We don't put _parameters on the queue so we either clone it or wait
-                # for it.
-                if not self._wait_for_clone:
+                # In the _parameters task, we need to resolve who is the resume leader.
+                self._is_resume_leader = False
+                resume_leader = None
+
+                if task_id_exists_already:
+                    # If the task id already exists, we need to check if current task is the resume leader in previous attempt.
+                    ds = self._flow_datastore.get_task_datastore(
+                        self.run_id, self.step, self.task_id
+                    )
+                    if not ds["_task_ok"]:
+                        raise MetaflowInternalError(
+                            "Externally cloned _parameters task did not succeed"
+                        )
+
+                    # Check if we should be the resume leader (maybe from previous attempt)
+                    if ds.has_metadata("_resume_leader", add_attempt=False):
```

Comment:
This is a bit tricky here -- there is a possible race where the resume leader is elected but never writes out the metadata which would mean that then no one is the resume leader. We may want to comment on this here but I think we can ignore it for now as it really shouldn't happen.

Reply:
I think it is fine here because it is under "task_id_exists_already=True", which means
1. We are in first attempt, since task_id_exists_already is true, we can not be leader anyways.
2. We are in second attempt, and we could be leader from writes in previous attempt (no race here)

Reply:
talked offline, added code to retry (to avoid race condition)